hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1407,43 +1382,7 @@ protected void cleanIfNecessary(BaseHoodieWriteClient writeClient, String instan
     writeClient.lazyRollbackFailedIndexing();
   }
 
-  /**
-   * Validates the timeline for both main and metadata tables to ensure compaction on MDT can be scheduled.
-   */
-  protected boolean validateTimelineBeforeSchedulingCompaction(Option<String> inFlightInstantTimestamp, String latestDeltaCommitTimeInMetadataTable) {
-    // we need to find if there are any inflights in data table timeline before or equal to the latest delta commit in metadata table.
-    // Whenever you want to change this logic, please ensure all below scenarios are considered.
-    // a. There could be a chance that latest delta commit in MDT is committed in MDT, but failed in DT. And so findInstantsBeforeOrEquals() should be employed
-    // b. There could be DT inflights after latest delta commit in MDT and we are ok with it. bcoz, the contract is, the latest compaction instant time in MDT represents
-    // any instants before that is already synced with metadata table.
-    // c. Do consider out of order commits. For eg, c4 from DT could complete before c3. and we can't trigger compaction in MDT with c4 as base instant time, until every
-    // instant before c4 is synced with metadata table.
-    List<HoodieInstant> pendingInstants = dataMetaClient.reloadActiveTimeline().filterInflightsAndRequested()
-        .findInstantsBeforeOrEquals(latestDeltaCommitTimeInMetadataTable).getInstants();
-
-    if (!pendingInstants.isEmpty()) {
-      checkNumDeltaCommits(metadataMetaClient, dataWriteConfig.getMetadataConfig().getMaxNumDeltacommitsWhenPending());
-      LOG.info(String.format(
-          "Cannot compact metadata table as there are %d inflight instants in data table before latest deltacommit in metadata table: %s. Inflight instants in data table: %s",
-          pendingInstants.size(), latestDeltaCommitTimeInMetadataTable, Arrays.toString(pendingInstants.toArray())));
-      return false;
-    }
-
-    // Check if there are any pending compaction or log compaction instants in the timeline.
-    // If pending compact/logCompaction operations are found abort scheduling new compaction/logCompaction operations.
-    Option<HoodieInstant> pendingLogCompactionInstant =
```

Comment:
Hi, @suryaprasanna , can you help confirm this part, in `HoodieLogCompactionPlanGenerator` we already skip scheduling compaction plan for any pending compaction/logcompaction file slices, so why we still need to cancel the scheduling the compaction here?

Reply:
> validateTimelineBeforeSchedulingCompaction method basically checks if there are inflight instants or pending compaction or logcompaction plan on the timeline. If any such instants are found it will stop compaction or logcompaction from happening.

We now can unlock this restriction because the MDT reader will filter out the invalid instants.

> It also makes sure only one pending plan is present on the metadata table's timeline. That way it is easy to handle various cases on metadata table.

Can it be in line with what we have for DT, because a pending compaction plan on MDT would block all the following-up scheduling of the compactions.

> Pending logcompaction check present in HoodieLogCompactionPlanGenerator class, will exclude file groups that are currently under logcompaction, that means other file groups or partitions in metadata can still create another logcompaction plan.

But there is no plan on the file slice that has pending plans (compaction/log_compaction), so the plan should be still valid?

> One can argue that the, we should be able to schedule multiple logcompaction plans on the timeline, but we have seen operational issues when running logcompaction and compaction on metadata using table services. So, it is better keep these checks.

I just wanna to make sure the strategy as simple and on par with the DT, so that the streaming use cases can also work well.

cc @codope for helping to address some of these argue points maybe.

Reply:
Finally discussed offline with @suryaprasanna and we are concinved:

1. The first part check for pending instants can be removed because we have NB-CC style file slicing, the instant time sequence does not really matter, only the completion time plays the role;
2. The second part check can be limited to only the log compaction scope.

Reply:
As discussed offline, please create unit test around the inflight scenario and necessary followup ticket to remove logcompaction.

Reply:
`testMetadataTableCompactionWithPendingInstants` cover this use case.

Reply:
I’m wondering if the condition of scheduling compaction based on log compaction should be put into the compaction logic in the write/table service client, instead of the MDT writer.  Is this specific to MDT?  The logic should also apply to DT around completion time.

Reply:
Currently log compaction is only used for MDT, we should move it out in the future, cc @suryaprasanna to address this in issue: https://issues.apache.org/jira/browse/HUDI-7533